### PR TITLE
Add Common CAPTCHA to Comments to Prevent Malicious Submissions, Update QQ Interface

### PR DESCRIPTION
**Pull request by smallxu038**
My website has received dozens of malicious comments, nearly 100 meaningless comments. The simple human verification of the theme is not real verification, so I added 4 common types of CAPTCHA and ensured that their style and functionality are compatible with the theme. QQ has updated its api, the old api cannot retrieve information, the updated api can retrieve avatars but cannot properly retrieve nicknames. Chinese nicknames will decode incorrectly, I have tried common encoding formats to no avail, and need others to find the correct encoding format.
cloudflare
reCAPTCHAV2
reCAPTCHAV3
mCAPTCHA

### CODE REVIEW

**Comment by mashirozx** [comment]
A suggestion: can you please show the captcha in a popup modal that only display after the user click the submit button? Anyway, a nice job you've done 👍

**Comment by smallxu038** [comment]
Sure, I have integrated the invisible mode of reCAPTCHA v2 into the theme. You can choose whether to enable the invisible mode in the backend. However, the keys for v2 invisible mode and normal mode are not interchangeable, and the invisible mode requires users to check multiple images, which may reduce the willingness to comment. If you need real invisible verification, you can consider enabling reCAPTCHA v3. The new changes have added v3 score settings in the theme settings.

File: footer.php [hunk 1]
```diff
@@ -159,5 +153,33 @@ class="aplayer"
...
+
+  // Now submit the form via AJAX
+  form.data('form-submitted', true); // Mark the form as submitted
+  form.trigger('submit'); // Trigger the form submit event
+}
+
+// Automatically execute reCAPTCHA when the page loads
+$(document).ready(function() {
+  grecaptcha.execute();
+});
```
Comment (mashirozx) [inline footer.php:181]: submit 前的校验应该在这里处理
https://github.com/mashirozx/sakura/blob/3.x/js/sakura-app.js#L1672

File: comments.php [hunk 2]
```diff
@@ -46,11 +46,29 @@
...
+                                        } elseif(akina_option('verification_type') == 'mCAPTCHA'){
+                                        $robot_comments = '<style>.widget__mcaptcha-brand-name { color:83d6ff} #mcaptcha__widget-container {height: 80px; width: 80%; background-color: #F9F9F9; border-radius: 5px; box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);} @media (prefers-color-scheme: dark) {#mcaptcha__widget-container {height: 80px; width: 80%; background-color: #505050; border-radius: 5px; box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);}} @media (prefers-color-scheme: light) {#mcaptcha__widget-container {height: 80px; width: 80%; background-color: #F9F9F9; border-radius: 5px; box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);}}</style><label class="siren-checkbox-label"><div id="mcaptcha__widget-container"></div><script charset="utf-8">let config = {widgetLink: new URL("'.akina_option('site_key').'"),};new mcaptchaGlue.default(config);</script></label>';
```
Comment (mashirozx) [inline comments.php:68]: 可以放到全局的样式表里

File: comments.php [hunk 3]
```diff
@@ -46,11 +46,29 @@
...
+                             data-callback="onRecaptchaSubmit"
+                             data-size="invisible">
+	                </div>
+			<?php }?>
 			<?php
 				$robot_comments = '';
 				if(comments_open()){
-					if(akina_option('norobot')) $robot_comments = '<label class="siren-checkbox-label"><input class="siren-checkbox-radio" type="checkbox" name="no-robot"><span class="siren-no-robot-checkbox siren-checkbox-radioInput"></span>'.__('I\'m not a robot', 'sakura').'</label>';
+					if(akina_option('norobot')) 
+					if (akina_option('verification_type') == 'CF Turnstile') {
```
Comment (mashirozx) [inline comments.php:61]: 建议单独放到一个 php 文件里 include 进来

**Comment by mashirozx** [comment]
如果还继续合并这边的话，把 captcha 相关的功能单独拉到一个分支里吧

File: comments.php (hunk 2 above)
Comment (smallxu038) [reply comments.php:68]: 其实这里不放也行，因为mCAPTCHA的验证码是一个透明的小部件，没有背景，这里是为他加上背景以符合主题风格。
不过为了标准，放进style.css确实是最合适的办法，我只是把我目前使用的代码直接提交过来了，这是我的疏忽。

**Comment by smallxu038** [comment]
> 如果还继续合并这边的话，把 captcha 相关的功能单独拉到一个分支里吧

我还发现我提交的代码中有一些问题，不过现在我需要先完成其他作业，稍晚我会提交修改后的代码到一个单独分支。

**Pull request closed without merging**